tasks/hierarchy.py

```diff
@@ -0,0 +1,540 @@
+from luigi import Task, Parameter, IntParameter, ListParameter
+from tasks.base_tasks import TempTableTask
+from tasks.meta import current_session
+from tasks.targets import ConstraintExistsTarget
+from tasks.targets import PostgresTarget, PostgresFunctionTarget
+from lib.logger import get_logger
+
+LOGGER = get_logger(__name__)
+
+
+class _CountryTask:
+    @property
+    def _country(self):
+        raise NotImplementedError('_CountryTask must define country()')
+
+
+class DenormalizedHierarchy(Task, _CountryTask):
```

Comment:
The division of tasks in this is quite nice. It should be able to override some tasks per-country to account for differences. In particular, it should be possible to override `HierarchyChildParent` for geometries where the hierachy can be determined by the id (hierarchical ids) and save us a lot of computing time :rainbow: